core/src/main/scala/chandu0101/scalajs/react/components/ReactTable.scala

```diff
@@ -36,272 +48,242 @@ object ReactTable {
       display.flex,
       flexDirection.column,
       boxShadow := "0 1px 3px 0 rgba(0, 0, 0, 0.12), 0 1px 2px 0 rgba(0, 0, 0, 0.24)",
-      media.maxWidth(740 px)(
-        boxShadow := "none"
-      )
+      media.maxWidth(740 px)(boxShadow := "none")
     )
 
-    val tableRow = style(
-      display.flex,
-      padding :=! "0.8rem",
-      &.hover.apply(
-        backgroundColor :=! "rgba(244, 244, 244, 0.77)"
-      ),
-      media.maxWidth(740 px)(
-        display.flex,
-        flexDirection.column,
-        textAlign.center,
-        boxShadow := "0 1px 3px grey",
-        margin(5 px)
-      ),
-      unsafeChild("div")(
-        flex := "1"
-      )
-    )
+    val tableRow = style(padding :=! "0.8rem",
+                         &.hover(backgroundColor :=! "rgba(244, 244, 244, 0.77)"),
+                         media.maxWidth(740 px)(boxShadow := "0 1px 3px grey", margin(5 px)))
 
     val tableHeader = style(fontWeight.bold, borderBottom :=! "1px solid #e0e0e0", tableRow)
 
     val settingsBar = style(display.flex, margin :=! "15px 0", justifyContent.spaceBetween)
 
     val sortIcon = styleF.bool(
-      ascending =>
+      ascending ⇒
         styleS(
-          &.after.apply(
-            fontSize(9 px),
-            marginLeft(5 px),
-            if (ascending) content := "'\\25B2'"
-            else content := "'\\25BC'"
-          )
-      ))
+          &.after(fontSize(9 px), marginLeft(5 px), if (ascending) { content := "'\\25B2'" } else {
+            content := "'\\25BC'"
+          })))
 
   }
 
   object DefaultStyle extends Style
 
-  case class State(
-      filterText: String,
-      offset: Int,
-      rowsPerPage: Int,
-      filteredModels: Vector[Model],
-      sortedState: Map[String, String]
-  )
+  type CellRenderer[T] = T ⇒ VdomNode
+
+  def DefaultCellRenderer[T]: CellRenderer[T] = { model =>
+    <.span(model.toString)
+  }
+  def EmailRenderer[T](fn: T => String): CellRenderer[T] = { t =>
+    val str = fn(t)
+    <.a(^.whiteSpace.nowrap, ^.href := s"mailto:${str}", str)
+  }
+  def OptionRenderer[T, B](defaultValue: String = "")(fn: T => Option[B]): CellRenderer[T] =
+    t => fn(t).fold(defaultValue)(_.toString)
```

Comment:
`toString` looks wrong. shouldn't you be using `CellRenderer[B]`? also `defaultValue` should be `VdomNode`?